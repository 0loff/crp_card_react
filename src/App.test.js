import { render, screen } from '@testing-library/react';
import App from './App';
import React from "react";

describe('App component', () => {
  it('render App component with fetched data',  async () => {
    render(<App />);

    expect(await screen.findByText(/Post 1/i)).toBeInTheDocument();
    expect(await screen.findByText(/Post 2/i)).toBeInTheDocument();
    expect(await screen.findByText(/Post 3/i)).toBeInTheDocument();

    expect(await screen.findByText('go to goole')).toHaveProperty('href')
    expect(await screen.findByText('go to yandex')).toHaveProperty('href')
    expect(await screen.findByText('go to bing')).toHaveProperty('href')
  });

  it('render App component without waiting fetched data', () => {
    const { asFragment } = render(<App />);

    expect(asFragment(<App />)).toMatchSnapshot();
  })
})




