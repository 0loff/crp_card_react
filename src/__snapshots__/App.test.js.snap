// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render App component without waiting fetched data 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="card_wrapper"
    >
      <div
        class="loader_wrapper"
      >
        <div
          class="loader"
        >
          <div />
          <div />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
